sdk/trace/span.go

```diff
@@ -134,12 +134,6 @@ func (s *span) End(options ...apitrace.EndOption) {
 			} else {
 				sd.EndTime = opts.EndTime
 			}
-			// Sampling check would be in the processor if the processor is used for exporting.
-			if s.spanContext.IsSampled() {
-				for e := range exp {
-					e.ExportSpan(sd)
-				}
-			}
```

Comment:
This check should be added to SimpleSpanProcessor and BatchedSpanProcessor.

Reply:
Sorry, I missed this comment. Wouldn't that change the behavior though? `OnEnd` is currently being executed without that check.
Also, `spanContext` isn't available from within the processor.

Reply:
I believe you are referring to change in behavior of SSP and BSP as neither of them have the check right now. The current behavior is incorrect as it exports everything instead of just the sampled spans. SpanData has SpanContext so it should be able to do the check.

Reply:
All right, I've added the check. I've been looking for ways to setup an unsampled SpanContext to no luck though. So I'm happy to get any pointers to a way to test it.